backend/src/main/java/org/eclipse/tractusx/puris/backend/common/ddtr/logic/util/DtrRequestBodyBuilder.java

```diff
@@ -0,0 +1,252 @@
+        ObjectNode externalSubjectIdObject = objectMapper.createObjectNode();
+        customerPartIdObject.set("externalSubjectId", externalSubjectIdObject);
+        externalSubjectIdObject.put("type", "ExternalReference");
+        ArrayNode keysArray = objectMapper.createArrayNode();
+        externalSubjectIdObject.set("keys", keysArray);
+        keysArray.add(getOwnReferenceObject());
+        keysArray.add(getReferenceObject(customerBpnl));
+        return customerPartIdObject;
+    }
+
+    private ObjectNode getReferenceObject(String bpnl) {
+        var refObject = objectMapper.createObjectNode();
+        refObject.put("type", "GlobalReference");
+        refObject.put("value", bpnl);
+        return refObject;
+    }
+
+    private ObjectNode getOwnReferenceObject() {
+        return getReferenceObject(variablesService.getOwnBpnl());
+    }
+
+
+    private JsonNode getDigitalTwinObject(List<ObjectNode> refNodes) {
+        var digitalTwinObject = objectMapper.createObjectNode();
+        digitalTwinObject.put("name", DIGITAL_TWIN_TYPE);
+        digitalTwinObject.put("value", "PartType");
+        var externalSubjectIdObject = objectMapper.createObjectNode();
+        digitalTwinObject.set("externalSubjectId", externalSubjectIdObject);
+        externalSubjectIdObject.put("type", "ExternalReference");
+        var keysArray = objectMapper.createArrayNode();
+        externalSubjectIdObject.set("keys", keysArray);
+        keysArray.add(getOwnReferenceObject());
+        for (var refNode : refNodes) {
+            keysArray.add(refNode);
+        }
+        return digitalTwinObject;
+    }
+
+    private JsonNode getItemStockSubmodelObject() {
+        var itemStockRequestSubmodelObject = objectMapper.createObjectNode();
+
+        itemStockRequestSubmodelObject.put("id", UUID.randomUUID().toString());
+        var semanticIdObject = objectMapper.createObjectNode();
+        itemStockRequestSubmodelObject.set("semanticId", semanticIdObject);
+        semanticIdObject.put("type", "ExternalReference");
+        var keysArray = objectMapper.createArrayNode();
+        semanticIdObject.set("keys", keysArray);
+        var keyObject = objectMapper.createObjectNode();
+        keysArray.add(keyObject);
+        keyObject.put("type", "GlobalReference");
+        keyObject.put("value", "urn:samm:io.catenax.item_stock:2.0.0#ItemStock");
+
+        var endpointsArray = objectMapper.createArrayNode();
+        itemStockRequestSubmodelObject.set("endpoints", endpointsArray);
+        var submodel3EndpointObject = objectMapper.createObjectNode();
+        endpointsArray.add(submodel3EndpointObject);
+        submodel3EndpointObject.put("interface", "SUBMODEL-3.0");
+        var protocolInformationObject = objectMapper.createObjectNode();
+        submodel3EndpointObject.set("protocolInformation", protocolInformationObject);
+        protocolInformationObject.put("href", variablesService.getEdcProtocolUrl());
```

Comment:
Done, although that's really not needed, because the proxy pull transfer-mechanism always provides that url alongside with the EDR Token. 
Since you always have to account for the possibiliy that the partner on the other side has registered a new data plane (with a new url) at his contol plane in the meantime, it's always wiser to only use the public data plane url that comes with the EDR Token.